src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -697,6 +697,11 @@ void MacroAssembler::call_VM_base(Register oop_result,
   // do the call, remove parameters
   MacroAssembler::call_VM_leaf_base(entry_point, number_of_arguments, &l);
 
+  // lr could be poisoned with PAC signature during throw_pending_exception
+  // if it was tail-call optimized by compiler, since lr is not callee-saved
+  // reload it with proper value
+  adr(lr, l);
```

Comment:
1. Instead of adr, using xpaclri would be better logically, and potentially faster. Plus on non-pac hardware it's just a NOP, so even better.
The AArch64 assembler does not yet have any pac instructions .... but I have already written that code. I'm happy to give you that patch, either for this PR or I can add a new PR (I hadn't done this yet because nothing would then use those instructions).

2. I'm unsure as to why the LR could be signed here. If the LR is signed, then how did the code jump back here without unsigning first? If a throw_pending_exception has been tail call optimised then it doesn't create a stack frame, and therefore shouldn't have signed the LR.   Blindly stripping the value in the caller feels very wrong from a PAC standpoint - the callee should be doing both signing and unsigning.

Reply:
Hello, the final callee which does return is pthread_jit_write_protect_np
it does pacibsp at the beginning and i was told it's intentional: 
>pthread_jit_write_protect_np now needs to push a frame in 11.4, so it PACs lr before spilling it and authenticates it before returning. 
>the shared cache in arm64 processes is arm64e and the B keys are enabled, allowing non-ABI-visible ptrauth in first-party libraries including libpthread.

This fix is more about saving LR before call and restoring it after call, as LR is not callee-saved.
So I think it's better to restore LR to proper state than stripping PAC signature in it. it's not even ARM64E  process here so I'm unsure if we should use some PAC-related asm code.

What you said is better be done as part of http://openjdk.java.net/jeps/8264131 in future.

Reply:
>and authenticates it before returning

Which means the pac signature will have been removed. So there is something more subtle going on. But that's probably irrelevant, because....

>LR is not callee-saved

Agreed. Which means we can't rely on it. In practice it always is restored.... except in this one case :) So, yes, adr is the correct thing to do here.

Ok, I'm happy with this change.

Reply:
[ This review was meant for the mainline patch, not jdk13 - I clicked on the wrong link. Either way, mainline one had been pushed anyway ]